markets/perps-market/contracts/modules/AsyncOrderSettlementModule.sol

```diff
@@ -157,6 +157,7 @@ contract AsyncOrderSettlementModule is IAsyncOrderSettlementModule, IMarketEvent
             .updatePositionData(runtime.accountId, newPosition);
         emit MarketUpdated(
             updateData.marketId,
+            fillPrice,
```

Comment:
Are you sure you want fillPrice from the order here?
This will be:
`(priceBeforeOrder  + priceAfterOrder) /2`
I would think to enable USD denominated OI calculations, you'd want the current price based on the current skew?

Reply:
You're right, we want the pyth price here.